src/filter.rs

```diff
@@ -76,3 +76,181 @@ impl Filter {
+        let range = ::rand::distributions::Range::new(0, 3);
+        let index = range.ind_sample(&mut rng);
+
+        match index {
+            0 => {
+                let structured_data =
+                    match ::structured_data::StructuredData::new(0,
+                                ::test_utils::Random::generate_random(), 0, vec![],
+                                vec![public_sign_key.clone()], vec![], Some(&secret_sign_key)) {
+                        Ok(structured_data) => structured_data,
+                        Err(error) => panic!("StructuredData error: {:?}", error),
+                };
+                return ::data::Data::StructuredData(structured_data)
+            },
+            1 => {
+                let type_tag = ::immutable_data::ImmutableDataType::Normal;
+                let immutable_data =
+                        ::immutable_data::ImmutableData::new(type_tag, generate_random_vec());
+                return ::data::Data::ImmutableData(immutable_data)
+            },
+            2 => {
+                let plain_data = ::plain_data::PlainData::new(
+                        ::test_utils::Random::generate_random(), generate_random_vec());
+                return ::data::Data::PlainData(plain_data)
+            },
+            _ => panic!("Unexpected index.")
+        }
+    }
+
+    // TODO Brian: Randomize Content and rename to random_routing_message.
+    fn arbtrary_routing_message(public_key: &::sodiumoxide::crypto::sign::PublicKey,
+                              secret_key: &::sodiumoxide::crypto::sign::SecretKey)
+            -> ::messages::RoutingMessage {
+        let from_authority =
+                generate_random_authority(::test_utils::Random::generate_random(), public_key);
+        let to_authority =
+                generate_random_authority(::test_utils::Random::generate_random(), public_key);
+        let data = generate_random_data(public_key, secret_key);
+        let content = ::messages::Content::ExternalRequest(::messages::ExternalRequest::Put(data));
+
+        ::messages::RoutingMessage {
+            from_authority: from_authority,
+            to_authority: to_authority,
+            content: content,
+        }
+    }
+
+    #[test]
+    fn filter_check_before_duration_end() {
+        let duration = ::time::Duration::seconds(3);
+        let mut filter = super::Filter::with_expiry_duration(duration);
+        let claimant = ::types::Address::Node(::test_utils::Random::generate_random());
+        let keys = ::sodiumoxide::crypto::sign::gen_keypair();
+        let routing_message = arbtrary_routing_message(&keys.0, &keys.1);
+        let signed_message =
+            ::messages::SignedMessage::new(claimant.clone(), routing_message.clone(), &keys.1);
+        let signed_message = signed_message.unwrap();
+
+        assert!(filter.check(&signed_message));
+        assert!(!filter.check(&signed_message));
+    }
+
+    #[test]
+    fn filter_check_after_duration_end() {
```

Comment:
maybe make the filter duration 1 ms, and sleep 3 ms in between the asserts ;) - but your test is valid too